account_move_budget/models/res_partner.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 ForgeFlow S.L. (http://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    budget_count = fields.Integer(
+        "# Budgets", compute="_compute_budget_count", compute_sudo=False
+    )
+
+    def _get_budget_ids(self):
+        self.ensure_one()
+        budget_lines = self.env["account.move.budget.line"].search(
+            [("partner_id", "=", self.id)]
+        )
+        return list(set(budget_lines.mapped("budget_id.id")))
```

Comment:
mapped() already removes dupplicates, so:
`return budget_lines.budget_id.mapped("id")
should be enought!